File: 14.js
// 14. Longest Common Prefix

// Write a function to find the longest common prefix string amongst an array of strings.
// If there is no common prefix, return an empty string "".

/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
  let shortest = 200;
  let ret = "";
  for(let i = 0; i<strs.length; i++) {
    if(strs[i].length < shortest) { shortest = strs[i].length; }
  }
  for(let i = 0; i<shortest; i++) {
      let current = strs[0].charAt(i);
      for(let j = 0; j<strs.length; j++) {
          if(strs[j].charAt(i)!==current) {
              return ret;
          }
      }
      ret+=current;
  }
  return ret;
};

File: 14.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        min_length = 201
        for s in strs:
            if len(s) < min_length:
                min_length = len(s)
        test_prefix = strs[0][0:min_length]
        for s in strs:
            if s[0:min_length] == test_prefix:
                continue
            else:
                while min_length > 0 and s[0:min_length] != test_prefix[0:min_length]:
                    min_length -= 1
                if min_length == 0:
                    return ""
                else:
                    test_prefix = test_prefix[0:min_length]
        return test_prefix
